cartographer/mapping/pose_extrapolator.cc

```diff
@@ -208,10 +209,12 @@ void PoseExtrapolator::AdvanceImuTracker(const common::Time time,
 
 Eigen::Quaterniond PoseExtrapolator::ExtrapolateRotation(
     const common::Time time) {
-  ImuTracker imu_tracker = *imu_tracker_;
-  AdvanceImuTracker(time, &imu_tracker);
+  if (time < extrapolating_imu_tracker_->time()) {
+    extrapolating_imu_tracker_.reset(new ImuTracker(*imu_tracker_));
```

Comment:
Did a `CHECK` not work? It would seem better to constrain usage if possible. It should be easy to ensure that we can extrapolate at the caller, and doing so has a performance benefit.